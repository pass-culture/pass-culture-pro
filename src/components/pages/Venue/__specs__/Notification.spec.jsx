import { mount } from 'enzyme'
import React from 'react'
import { NavLink, Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import NotificationMessage from '../Notification'

describe('src | components | pages | Venue | Notification', () => {
  const props = {
    dispatch: jest.fn(),
    venueId: 'TY',
    offererId: 'FT',
  }

  it('should display a succes messsage when venue is created', () => {
    // given
    const history = createBrowserHistory()

    // when
    const wrapper = mount(
      <Router history={history}>
        <NotificationMessage {...props} />
      </Router>
    )
    const textMessage = wrapper.find('p')
    const navLink = wrapper.find(NavLink)

    // then
    expect(textMessage.text()).toBe(
      'Lieu créé. Vous pouvez maintenant y créer une offre, ou en importer automatiquement. '
    )
    expect(navLink.prop('to')).toBe('/offres/creation?lieu=TY&structure=FT')
  })
})
