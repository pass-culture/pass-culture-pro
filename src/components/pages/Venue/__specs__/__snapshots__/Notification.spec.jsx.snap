// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Venue | Notification should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BuildNotificationMessage
    dispatch={[MockFunction]}
    offererId="FT"
    venueId="TY"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Lieu créé. Vous pouvez maintenant y ",
        <NavLink
          activeClassName="active"
          aria-current="page"
          onClick={[Function]}
          to="/offres/creation?lieu=TY&structure=FT"
        >
          créer une offre
        </NavLink>,
        ", ou en importer automatiquement. ",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Lieu créé. Vous pouvez maintenant y ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "créer une offre",
          "onClick": [Function],
          "to": "/offres/creation?lieu=TY&structure=FT",
        },
        "ref": null,
        "rendered": "créer une offre",
        "type": [Function],
      },
      ", ou en importer automatiquement. ",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Lieu créé. Vous pouvez maintenant y ",
          <NavLink
            activeClassName="active"
            aria-current="page"
            onClick={[Function]}
            to="/offres/creation?lieu=TY&structure=FT"
          >
            créer une offre
          </NavLink>,
          ", ou en importer automatiquement. ",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Lieu créé. Vous pouvez maintenant y ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "créer une offre",
            "onClick": [Function],
            "to": "/offres/creation?lieu=TY&structure=FT",
          },
          "ref": null,
          "rendered": "créer une offre",
          "type": [Function],
        },
        ", ou en importer automatiquement. ",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
